### PR TITLE
Corrected issue where blank urls resulted in spurious returns to KeepassHTTP

### LINKED ISSUE #1017: KeePass returning unrelated credentials over keepasshttp
Hello, since the last AUR update, when I go on a random unknown site with a logon field, I get promoted to grant chromeIPass access to 7 account credentials.
It is always the same accounts but I see no pattern into why.

When on a site where I have credentials saved it works as expected. (although accounts on the same domain with different paths are also shown now)

Did something in the search/filtering from keepasshttp change?

## Expected Behavior
Nothing at all should happen, its not the correct site for the credentials.

## Current Behavior
Prompt opens to grant access to 7 unrelated accounts.

## Debug Info
Version | 2.2.0.r94.gbe312bbe-1
KeePassXC - Version 2.2.0
Revision: be312bbe4f7115d411893a07cf546b7eff9c65c1

Libraries:
- Qt 5.9.1
- libgcrypt 1.8.1

Operating system: Manjaro Linux
CPU architecture: x86_64
Kernel: linux 4.13.4-1-MANJARO

Enabled extensions:
- KeePassHTTP
- Auto-Type
- YubiKey

### PR DESCRIPTION
**Pull request by droidmonkey**
## Description

Fixes #1017 

## How has this been tested?

Manually with debugger

## Types of changes

- ✅ Bug fix (non-breaking change which fixes an issue)

## Checklist:

- ✅ I have read the **CONTRIBUTING** document. **[REQUIRED]**
- ✅ My code follows the code style of this project. **[REQUIRED]**
- ✅ All new and existing tests passed. **[REQUIRED]**
- ✅ I have compiled and verified my code with `-DWITH_ASAN=ON`. **[REQUIRED]**

### CODE REVIEW

**Comment by yan12125** [comment]
How about adding a test case? For example:
```Diff
diff --git a/tests/TestEntry.cpp b/tests/TestEntry.cpp
index 4d34cf31..fe37ca22 100644
--- a/tests/TestEntry.cpp
+++ b/tests/TestEntry.cpp
@@ -19,6 +19,7 @@
 
 #include <QTest>
 
+#include "config-keepassx-tests.h"
 #include "core/Entry.h"
 #include "crypto/Crypto.h"
 
@@ -130,3 +131,12 @@ void TestEntry::testClone()
 
     delete entryOrg;
 }
+
+void TestEntry::testResolveUrl()
+{
+#ifdef WITH_XC_HTTP
+    Entry* entry = new Entry();
+    QCOMPARE(entry->resolveUrl(""), QString(""));
+    delete entry;
+#endif
+}
diff --git a/tests/TestEntry.h b/tests/TestEntry.h
index 0c97c0b9..3f6d20ee 100644
--- a/tests/TestEntry.h
+++ b/tests/TestEntry.h
@@ -31,6 +31,7 @@ private slots:
     void testHistoryItemDeletion();
     void testCopyDataFrom();
     void testClone();
+    void testResolveUrl();
 };
 
 #endif // KEEPASSX_TESTENTRY_H
```
testResolveUrl() should have more tests. This is just a proof-of-concept.

**Comment by droidmonkey** [comment]
After reading the function a couple of times I don't like how the non-HTTP case returns an empty string. Shouldn't we just pass-through the input string?

**Comment by droidmonkey** [comment]
OK, so after doing a little test-driven-development I found that the resolveUrl function was completely broken in a variety of cases. I corrected those and wrote many test cases for it to be run against. I also found that the WITH_XC_HTTP guards were completely unnecessary as QUrl is usable regardless of the HTTP presence.

File: tests/TestEntry.cpp [hunk 1]
```diff
@@ -130,3 +131,28 @@ void TestEntry::testClone()
 
     delete entryOrg;
 }
+
+void TestEntry::testResolveUrl()
+{
+    Entry* entry = new Entry();
```
Comment (yan12125) [inline tests/TestEntry.cpp]: This should be ```delete```d? Or maybe make resolveUrl a static function.

Comment (droidmonkey) [reply tests/TestEntry.cpp]: It doesn't matter in a test. Technically it should be but it really doesn't matter.

Comment (droidmonkey) [reply tests/TestEntry.cpp]: ok made the change 😝

Comment (phoerious) [reply tests/TestEntry.cpp]: QUrl may be usable, but WITH_XC_HTTP is also for those people who don't like networking functionality inside their password manager's code.

Comment (droidmonkey) [reply tests/TestEntry.cpp]: There is no networking code in here.

Comment (phoerious) [reply tests/TestEntry.cpp]: True. I though, QUrl would also do some resolver work. But I guess I've just been doing to much Java as of late. :-)

Comment (TheZ3ro) [reply tests/TestEntry.cpp]: Was thinking the same thing when I first write that piece of code :D

**Pull request merged**